service/payment_test.go

```diff
@@ -173,3 +220,153 @@ func TestUnitCreatePaymentSession(t *testing.T) {
+		w := httptest.NewRecorder()
+		mockPaymentService.GetPaymentSession(w, req)
+		So(w.Code, ShouldEqual, 500)
+	})
+
+	Convey("Amount mismatch", t, func() {
+		mock := dao.NewMockDAO(mockCtrl)
+		mockPaymentService := createMockPaymentService(mock, cfg)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(models.PaymentResource{Amount: "100", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		req, err := http.NewRequest("Get", "", nil)
+		So(err, ShouldBeNil)
+		req.Body = ioutil.NopCloser(bytes.NewReader(reqBody))
+		q := req.URL.Query()
+		q.Add(":payment_id", "1234")
+		req.URL.RawQuery = q.Encode()
+		w := httptest.NewRecorder()
+		httpmock.Activate()
+		defer httpmock.DeactivateAndReset()
+		costArray := []models.CostResource{{Amount: "99"}}
+		jsonResponse, _ := httpmock.NewJsonResponder(200, costArray)
+		httpmock.RegisterResponder("GET", "http://dummy-resource", jsonResponse)
+		mockPaymentService.GetPaymentSession(w, req)
+		So(w.Code, ShouldEqual, 403)
+	})
+
+	Convey("Get Payment session - success - Single cost", t, func() {
+		mock := dao.NewMockDAO(mockCtrl)
+		mockPaymentService := createMockPaymentService(mock, cfg)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(models.PaymentResource{Amount: "10", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		req, err := http.NewRequest("Get", "", nil)
+		So(err, ShouldBeNil)
+		req.Body = ioutil.NopCloser(bytes.NewReader(reqBody))
+		q := req.URL.Query()
+		q.Add(":payment_id", "1234")
+		req.URL.RawQuery = q.Encode()
+		w := httptest.NewRecorder()
+		httpmock.Activate()
+		defer httpmock.DeactivateAndReset()
+		costArray := []models.CostResource{{Amount: "10"}}
+		jsonResponse, _ := httpmock.NewJsonResponder(200, costArray)
+		httpmock.RegisterResponder("GET", "http://dummy-resource", jsonResponse)
+		mockPaymentService.GetPaymentSession(w, req)
+		So(w.Code, ShouldEqual, 200)
+	})
+
+	Convey("Get Payment session - success - Multiple costs", t, func() {
+		mock := dao.NewMockDAO(mockCtrl)
+		mockPaymentService := createMockPaymentService(mock, cfg)
+		mock.EXPECT().GetPaymentResource(gomock.Any()).Return(models.PaymentResource{Amount: "23", Links: models.Links{Resource: "http://dummy-resource"}}, nil)
+		req, err := http.NewRequest("Get", "", nil)
+		So(err, ShouldBeNil)
+		req.Body = ioutil.NopCloser(bytes.NewReader(reqBody))
+		q := req.URL.Query()
+		q.Add(":payment_id", "1234")
+		req.URL.RawQuery = q.Encode()
+		w := httptest.NewRecorder()
+		httpmock.Activate()
+		defer httpmock.DeactivateAndReset()
+		costArray := []models.CostResource{{Amount: "10"}, {Amount: "13"}}
+		jsonResponse, _ := httpmock.NewJsonResponder(200, costArray)
+		httpmock.RegisterResponder("GET", "http://dummy-resource", jsonResponse)
+		mockPaymentService.GetPaymentSession(w, req)
+		So(w.Code, ShouldEqual, 200)
+	})
+}
+
+func TestUnitGetTotalAmount(t *testing.T) {
+	Convey("Get Total Amount", t, func() {
```

Comment:
I'm assuming we'll never have negative costs (e.g. discounts) but I figured I'd ask. If we did, would it be worth adding unit test coverage for this case?